### ISSUE TITLE
`random_state` parameter does not conform to sklearn interface

### ISSUE THREAD

**Issue by dpaetzel** [xcsf-dev/xcsf#96]
This is a minor nuisance but I though I'd document it anyways.

The [sklearn interface](https://scikit-learn.org/stable/developers/develop.html#random-numbers) wants an estimator's `random_state` argument to be `None` by default (corresponding to not seeding the RNG) and typically to accept values >= 0 as valid seeds.

However,

```python
In [3]: xcsf.XCS().get_params()["random_state"]
Out[3]: 0
```

(should return `None`) as well as

```python
In [4]: xcsf.XCS(random_state=None)
Error: unable to import parameter: random_state
```

(should work).

**Comment by dpaetzel** [comment]
(I should mention that the latter of the two actually seems to kill the Python process without being catchable via exception handling.)

**Comment by rpreen** [comment]
I agree ideally `None` would be better than using the value `0` which is the current equivalent. `random_state` only accepts `int` values at present. This will require mapping `None` to something like `-1` internally and then back again, and using negative numbers to set the seed with the current time instead of `0`.
